packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -0,0 +1,134 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import jakarta.annotation.PostConstruct;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import software.uncharted.terarium.hmiserver.annotations.IgnoreRequestLogging;
+import software.uncharted.terarium.hmiserver.models.dataservice.document.DocumentAsset;
+import software.uncharted.terarium.hmiserver.models.dataservice.model.Model;
+import software.uncharted.terarium.hmiserver.models.task.TaskRequest;
+import software.uncharted.terarium.hmiserver.models.task.TaskResponse;
+import software.uncharted.terarium.hmiserver.models.task.TaskStatus;
+import software.uncharted.terarium.hmiserver.security.Roles;
+import software.uncharted.terarium.hmiserver.service.TaskResponseHandler;
+import software.uncharted.terarium.hmiserver.service.TaskService;
+import software.uncharted.terarium.hmiserver.service.data.DocumentAssetService;
+import software.uncharted.terarium.hmiserver.service.data.ModelService;
+
+@RequestMapping("/gollm")
+@RestController
+@Slf4j
+@RequiredArgsConstructor
+public class GoLLMController {
+
+	final private ObjectMapper objectMapper;
+	final private TaskService taskService;
+	final private DocumentAssetService documentAssetService;
+	final private ModelService modelService;
+
+	final private String MODEL_CARD_SCRIPT = "gollm:model_card";
+
+	@Data
+	private static class ModelCardInput {
+		@JsonProperty("research_paper")
+		String researchPaper;
+	};
+
+	@Data
+	private static class ModelCardProperties {
+		UUID modelId;
+		UUID documentId;
+	};
+
+	@PostConstruct
+	void init() {
+		taskService.addResponseHandler(MODEL_CARD_SCRIPT, getModelCardResponseHandler());
+	}
+
+	private TaskResponseHandler getModelCardResponseHandler() {
+		TaskResponseHandler handler = new TaskResponseHandler();
+		handler.onSuccess((TaskResponse resp) -> {
+			ModelCardProperties props = resp.getAdditionalProperties(ModelCardProperties.class);
+			log.info("Writing model card to database for model {}", props.getModelId());
+			try {
+				Model model = modelService.getModel(props.getModelId())
+						.orElseThrow();
+				model.setModelCard(new String(resp.getOutput()));
+				modelService.updateModel(model);
+			} catch (IOException e) {
+				log.error("Failed to write model card to database", e);
+			}
+		});
+		return handler;
+	}
+
+	@PostMapping("/model_card")
+	@Secured(Roles.USER)
+	public ResponseEntity<TaskResponse> createModelCardTask(
+			@RequestParam(name = "document-id", required = true) UUID documentId,
+			@RequestParam(name = "model-id", required = true) UUID modelId)
+			throws JsonProcessingException, IOException {
```

Comment:
Rather than throwing these for the client to deal with do we want to catch them and then throw a `ResponseStatusException` with a `INTERNAL_SERVER_ERROR` or whatever seems appropriate?

Reply:
Fixed!